server/src/main/java/org/spine3/server/entity/storage/ProtoToJavaMapper.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Any;
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.BytesValue;
+import com.google.protobuf.DoubleValue;
+import com.google.protobuf.FloatValue;
+import com.google.protobuf.Int32Value;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.Message;
+import com.google.protobuf.StringValue;
+import com.google.protobuf.UInt32Value;
+import com.google.protobuf.UInt64Value;
+import org.spine3.protobuf.AnyPacker;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A utility for converting the {@linkplain Message Protobuf Messages} (in form of {@link Any}) into
+ * arbitrary {@linkplain Object Java Objects}.
+ *
+ * <p>Currently, the supported types are:
+ * <ul>
+ *     <li>{@link Message} - converted via {@link AnyPacker};
+ *     <li>Java primitives - the passed {@link Any} is unpacked into one of types {@code Int32Value,
+ *     Int64Value, UInt32Value, UInt64Value, FloatValue, DoubleValue, BoolValue, StringValue,
+ *     BytesValue} and then into the corresponding Java type, either a primitive value, or
+ *     {@code String} of {@link ByteString}.
+ * </ul>
+ *
+ * @author Dmytro Dashenkov
+ */
+// TODO:2017-04-21:dmytro.dashenkov: Maybe we should move this to client code to make this util reusable.
+final class ProtoToJavaMapper {
```

Comment:
That belongs to `client` / `org.spine3.protobuf`. And then it should probably be `@Internal` and get used only from `core-java` and `tools` projects.